pkg/kubelet/prober/worker.go

```diff
@@ -153,10 +153,15 @@ func (w *worker) run() {
 	probeTickerPeriod := time.Duration(w.spec.PeriodSeconds) * time.Second
 
 	// If kubelet restarted the probes could be started in rapid succession.
-	// Let the worker wait for a random portion of tickerPeriod before probing.
+	// Let the worker wait for a random portion of tickerPeriod before probing,
+	// up to a maximum of 1 minute.
 	// Do it only if the kubelet has started recently.
 	if probeTickerPeriod > time.Since(w.probeManager.start) {
-		time.Sleep(time.Duration(rand.Float64() * float64(probeTickerPeriod)))
+		maxSleepDuration := time.Minute
```

Comment:
/hold

we should make these changes with a wider consensus, adding magic timeouts deep in the code is confusing.
There is also the other angle, this hitter mechanism is still more confusing, is it documented, is it tested?

we should get this better covered

Reply:
I hear you @aojea.

1. I'll add some tests around the new functionality.
2. Do you want me to add some small section at https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes, explaining about possible probing delay when kubelet restarts?
3. I'll get rid of the sleep command, in the favor of a select statement, similar to the one used in probeLoop.

Please do let me know if you have other suggestions